enhancements/cluster-logging/loki-gateway-configuration.md

```diff
@@ -0,0 +1,517 @@
+---
+title: loki-lokistack-gateway-configuration
+
+authors:
+  - "@sasagarw"
+
+reviewers:
+  - "@periklis"
+
+approvers:
+  - "@periklis"
+  - "@igor-karpukhin"
+
+creation-date: 2021-08-27
+last-updated: 2021-09-24
+status: implementable
+see-also: []
+replaces: []
+superseded-by: []
+---
+
+# Loki Operator: LokiStack CR extension for the gateway configuration
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+LokiStack Gateway is a component deployed as part of Loki Operator. It secure access on the loki cluster distributor and query-frontend endpoint via an OAuth/OIDC-handling gateway.
+
+This proposal seeks to provide OIDC configuration for the LokiStack gateway to integrate Loki access to the org's OIDC provider.
+
+Note: We are simply providing an API to reconfigure an existing implementation. Most of this context does not need to be implemented as it is already satisfied by [observatorium-api](https://github.com/observatorium/api).
+
+## Motivation
+
+### Goals
+
+The goals of this proposal are:
+
+* Provide changes to API to accept OIDC provider in LokiStack CR
+* Make OIDC configuration optional and auto-configuration on OCP
+* Secrets/Credentials/SA-Keys should be provided via a k8s secret that the LokiStack admin provides upfront
+
+We will be successful when our LokiStack CR is able to accept OIDC provider and able to configure LokiStack Gateway
+
+### Non-Goals
+
+### Risks and Mitigations
+
+NA
+
+### User Stories
+
+* As a loki cluster admin, I want The LokiStack CR be able to accept OIDC configuration.
+
+* As a cluster admin, I want the Loki Operator be able to reconcile correctly.
+
+## Design Details
+
+For configuring the LokiStack Gateway, we need information about oidc provider and opa.
```

Comment:
Identifier always in capital letters, i.e. `OIDC`, `OPA`. AFAICS OPA is mentioned here the first time and deserves a full naming as `OpenPolicyAgent (OPA)`